t/unit/backends/test_azureblockblob.py

```diff
@@ -0,0 +1,125 @@
+    def setup(self):
+        self.url = (
+            "azureblockblob://"
+            "DefaultEndpointsProtocol=protocol;"
+            "AccountName=name;"
+            "AccountKey=key;"
+            "EndpointSuffix=suffix")
+
+        self.backend = AzureBlockBlobBackend(
+            app=self.app,
+            url=self.url)
+
+    def test_missing_third_party_sdk(self):
+        azurestorage = azureblockblob.azurestorage
+        try:
+            azureblockblob.azurestorage = None
+            with pytest.raises(ImproperlyConfigured):
+                AzureBlockBlobBackend(app=self.app, url=self.url)
+        finally:
+            azureblockblob.azurestorage = azurestorage
+
+    def test_bad_connection_url(self):
+        with pytest.raises(ImproperlyConfigured):
+            AzureBlockBlobBackend._parse_url("azureblockblob://")
+
+        with pytest.raises(ImproperlyConfigured):
+            AzureBlockBlobBackend._parse_url("")
+
+    @patch(MODULE_TO_MOCK + ".BlockBlobService")
+    def test_create_client(self, mock_blob_service_factory):
+        mock_blob_service_instance = Mock()
+        mock_blob_service_factory.return_value = mock_blob_service_instance
+        backend = AzureBlockBlobBackend(app=self.app, url=self.url)
+
+        # ensure container gets created on client access...
+        assert mock_blob_service_instance.create_container.call_count == 0
+        assert backend._client is not None
+        assert mock_blob_service_instance.create_container.call_count == 1
+
+        # ...but only once per backend instance
+        assert backend._client is not None
+        assert mock_blob_service_instance.create_container.call_count == 1
+
+    def test_get(self):
+        with patch_client() as mock_client:
+            self.backend.get(b"mykey")
+
+            mock_client.get_blob_to_text.assert_has_calls(
+                [call("celery", "mykey")])
+
+    def test_get_missing(self):
+        with patch_client() as mock_client:
+            mock_client.get_blob_to_text.side_effect = \
+                azureblockblob.AzureMissingResourceHttpError("Missing", 404)
+
+            assert self.backend.get(b"mykey") is None
+
+    def test_set(self):
+        with patch_client() as mock_client:
+            self.backend.set(b"mykey", "myvalue")
+
+            mock_client.create_blob_from_text.assert_called_once_with(
+                "celery", "mykey", "myvalue")
+
+    def test_mget(self):
+        with patch_client() as mock_client:
+            keys = [b"mykey1", b"mykey2"]
+
+            self.backend.mget(keys)
+
+            mock_client.get_blob_to_text.assert_has_calls(
+                [call("celery", "mykey1"),
+                 call("celery", "mykey2")])
+
+    def test_delete(self):
+        with patch_client() as mock_client:
+            self.backend.delete(b"mykey")
+
+            mock_client.delete_blob.assert_called_once_with(
+                "celery", "mykey")
+
+    @skip.unless_environ("AZUREBLOCKBLOB_URL")
```

Comment:
If I understand correctly, this test case probably belongs to the integration test suite.

Reply:
I'm having some issues running the integration tests locally via docker-compose so I've made a change and pushed it up to Travis in 9ba725d to see if the move of the test was successful. I'll tag you on the PR when everything is ready to be re-reviewed.

Reply:
Alright, confirmed that integration tests ran and the new backend test executed and passed. Sample CI job log: https://travis-ci.org/celery/celery/jobs/372769687